From 671583ab95214ace32933f6798f8801e57278584 Mon Sep 17 00:00:00 2001
From: Julian Scheel <julian@jusst.de>
Date: Mon, 14 May 2012 16:44:33 +0200
Subject: [PATCH] tsdemux: Update firstbuffer timestamp when it is 0.

As 0 is a valid timestamp the current detection will allow 0-timestamps in
firstbuffer packets. Actually this breaks streams where the pts is not read in
the first packet. This was seen on cablecom IPTV streams.
---
 gst/mpegtsdemux/tsdemux.c |    3 ++-
 1 Datei geändert, 2 Zeilen hinzugefügt(+), 1 Zeile entfernt(-)

diff --git a/gst/mpegtsdemux/tsdemux.c b/gst/mpegtsdemux/tsdemux.c
index 96eb75c..8f71c31 100644
--- a/gst/mpegtsdemux/tsdemux.c
+++ b/gst/mpegtsdemux/tsdemux.c
@@ -1542,7 +1542,8 @@ gst_ts_demux_push_pending_data (GstTSDemux * demux, TSDemuxStream * stream)
   GST_DEBUG_OBJECT (stream->pad, "stream->pts %" GST_TIME_FORMAT,
       GST_TIME_ARGS (stream->pts));
   if (GST_CLOCK_TIME_IS_VALID (stream->pts)
-      && !GST_CLOCK_TIME_IS_VALID (GST_BUFFER_TIMESTAMP (firstbuffer)))
+      && (!GST_CLOCK_TIME_IS_VALID (GST_BUFFER_TIMESTAMP (firstbuffer))
+          || GST_BUFFER_TIMESTAMP (firstbuffer) == 0))
     GST_BUFFER_TIMESTAMP (firstbuffer) =
         mpegts_packetizer_pts_to_ts (packetizer, stream->pts);
 
-- 
1.7.10.1

